client/state/purchases/selectors.js

```diff
@@ -0,0 +1,7 @@
+// External dependencies
+import assign from 'lodash/assign';
+import find from 'lodash/find';
+
+export const getByPurchaseId = ( state, id ) => (
```

Comment:
Well, there is only one criteria, so I would expect `by` to appear only once, but I might be misunderstanding the ambiguity of `getByPurchaseId`. The reason I wouldn't use `getPurchaseById` is that you can get purchase by the user or site ID as well.